contracts/v2/accounts/default/implementation_default.sol

```diff
@@ -20,6 +22,31 @@ contract Constants is Variables {
     // The Account Module Version.
     uint256 public constant version = 2;
 
+    // constants for EIP712 values
+    string public constant DOMAIN_SEPARATOR_NAME = "DeFi-Smart-Wallet";
+    string public constant DOMAIN_SEPARATOR_VERSION = "2.0.0";
+
+    // hashed EIP712 values
+    bytes32 internal constant DOMAIN_SEPARATOR_NAME_HASHED = keccak256(bytes(DOMAIN_SEPARATOR_NAME));
+    bytes32 internal constant DOMAIN_SEPARATOR_VERSION_HASHED = keccak256(bytes(DOMAIN_SEPARATOR_VERSION));
+
+    bytes32 internal constant TYPE_HASH =
+        keccak256("EIP712Domain(string name,string version,uint256 chainId,address verifyingContract)");
+
+    // EIP712 typehash for signed hashes used for EIP1271 (`isValidSignature()`)
+    bytes32 public constant EIP1271_TYPE_HASH = keccak256("DSA(bytes32 hash)");
+
+    // "magic value" according to EIP1271 https://eips.ethereum.org/EIPS/eip-1271#specification
+    bytes4 internal constant EIP1271_MAGIC_VALUE = 0x1626ba7e;
+
+    function getChainID() internal pure returns (uint256) {
```

Comment:
this is also available as block.chainId in solidity 0.8.0, but this is targeted for 0.7 right?

Reply:
Yess! DSA is with 0.7.0

Reply:
also a possibility is writing it to an immutable in constructor. But doesn't matter either way works